File: bench/ai/tasks/csv_field/starter.cpp
// Prints field k of every row of a comma-separated file.
#include <cstdlib>
#include <iostream>
#include <iterator>
#include <string>
#include <vector>

// Writes field k of the row text[lo, hi), where hi is the row's newline or the end of the input.
static void print_field(const std::vector<char>& text, std::size_t lo, std::size_t hi, std::size_t k) {
  std::size_t start = lo;
  for (std::size_t f = 0; f < k; ++f) {
    while (start < hi && text[start] != ',') ++start;
    if (start == hi) {
      std::cout << "none\n";
      return;
    }
    ++start;  // past the comma
  }
  std::size_t end = start;
  while (text[end] != ',' && text[end] != '\n') ++end;
  std::cout.write(text.data() + start, static_cast<std::streamsize>(end - start));
  std::cout << '\n';
}

int main() {
  std::vector<char> text((std::istreambuf_iterator<char>(std::cin)), std::istreambuf_iterator<char>());
  std::size_t pos = 0;
  while (pos < text.size() && text[pos] != '\n') ++pos;
  std::size_t k = std::strtoul(std::string(text.begin(), text.begin() + pos).c_str(), nullptr, 10);
  pos += 1;  // past the first line's newline
  while (pos < text.size()) {
    std::size_t hi = pos;
    while (hi < text.size() && text[hi] != '\n') ++hi;
    print_field(text, pos, hi, k);
    pos = hi + 1;
  }
  return 0;
}

File: bench/ai/tasks/csv_field/starter.rs
// Prints field k of every row of a comma-separated file.
use std::io::{self, Read, Write};

// Writes field k of the row text[lo..hi], where hi is the row's newline or the end of the input.
fn print_field(out: &mut impl Write, text: &[u8], lo: usize, hi: usize, k: usize) {
    let mut start = lo;
    for _ in 0..k {
        while start < hi && text[start] != b',' {
            start += 1;
        }
        if start == hi {
            out.write_all(b"none\n").unwrap();
            return;
        }
        start += 1; // past the comma
    }
    let mut end = start;
    while text[end] != b',' && text[end] != b'\n' {
        end += 1;
    }
    out.write_all(&text[start..end]).unwrap();
    out.write_all(b"\n").unwrap();
}

fn main() {
    let mut text = Vec::new();
    io::stdin().read_to_end(&mut text).unwrap();
    let mut out = io::stdout().lock();
    let mut pos = text.iter().position(|&b| b == b'\n').unwrap_or(text.len());
    let k: usize = std::str::from_utf8(&text[..pos]).unwrap().trim().parse().unwrap();
    pos += 1; // past the first line's newline
    while pos < text.len() {
        let mut hi = pos;
        while hi < text.len() && text[hi] != b'\n' {
            hi += 1;
        }
        print_field(&mut out, &text, pos, hi, k);
        pos = hi + 1;
    }
}
